root/static/scripts/common/immutable-entities.js

```diff
@@ -37,7 +37,7 @@ const reduceArtistCredit = ac => ac.names.reduce(reduceName, '');
 const isComplexArtistCredit = function (ac) {
   const firstName = ac.names.get(0);
   if (firstName && hasArtist(firstName)) {
-     return firstName.artist.name !== reduceArtistCredit(ac);
+     return firstName.artist.name !== reduceArtistCredit(ac) || !nonEmpty(firstName.name);
```

Comment:
Done in https://github.com/metabrainz/musicbrainz-server/pull/468/commits/37dd5949ef47d7606c78f2b3b959d0b8d31c272f